samples/astarte_app/src/config_file.c

```diff
@@ -0,0 +1,242 @@
+    if ((snprintf_rc < 0) || (snprintf_rc >= ARRAY_SIZE(out_cfg->wifi_ssid))) {
+        LOG_ERR("Error extracting the WiFi SSID from the parsed json.");
+        return -1;
+    }
+    snprintf_rc = snprintf(out_cfg->wifi_pwd, ARRAY_SIZE(out_cfg->wifi_pwd), "%s", wifi_pwd);
+    if ((snprintf_rc < 0) || (snprintf_rc >= ARRAY_SIZE(out_cfg->wifi_pwd))) {
+        LOG_ERR("Error extracting the WiFi password from the parsed json.");
+        return -1;
+    }
+#else
+    ARG_UNUSED(wifi_ssid);
+    ARG_UNUSED(wifi_pwd);
+#endif
+    return 0;
+}
+#if defined(CONFIG_CONFIG_FROM_FLASH)
+static int read_configuration_file(
+    char fname[static MAX_PATH_LEN], char fcontent[static MAX_CONFIG_FILE_SIZE])
+{
+    struct fs_file_t file;
+    int rc, ret;
+
+    fs_file_t_init(&file);
+    rc = fs_open(&file, fname, FS_O_READ);
+    if (rc < 0) {
+        LOG_ERR("FAIL: open %s: %d", fname, rc);
+        return rc;
+    }
+
+    rc = fs_read(&file, fcontent, MAX_CONFIG_FILE_SIZE);
+    if (rc <= 0) {
+        LOG_ERR("FAIL: read %s: [rd:%d]", fname, rc);
+        goto out;
+    }
+
+out:
+    ret = fs_close(&file);
+    if (ret < 0) {
+        LOG_ERR("FAIL: close %s: %d", fname, ret);
+        return ret;
+    }
+
+    return rc;
+}
+
+static int parse_configuration_file(char *fcontent, struct sample_config *out_cfg)
+{
+    // Define the structure of the expected JSON file
+    struct full_json_s
+    {
+        const char *deviceID;
+        const char *credentialSecret;
+        const char *wifiSsid;
+        const char *wifiPassword;
+    };
+    // Create the descriptors list containing each element of the struct
+    static const struct json_obj_descr full_json_descr[] = {
+        JSON_OBJ_DESCR_PRIM(struct full_json_s, deviceID, JSON_TOK_STRING),
+        JSON_OBJ_DESCR_PRIM(struct full_json_s, credentialSecret, JSON_TOK_STRING),
+        JSON_OBJ_DESCR_PRIM(struct full_json_s, wifiSsid, JSON_TOK_STRING),
+        JSON_OBJ_DESCR_PRIM(struct full_json_s, wifiPassword, JSON_TOK_STRING),
+    };
+    // This is only for the outside level, nested elements should be checked individually.
+    int expected_return_code = (1U << (size_t) ARRAY_SIZE(full_json_descr)) - 1;
+    // Parse the json into a structure
+    struct full_json_s parsed_json
+        = { .deviceID = NULL, .credentialSecret = NULL, .wifiSsid = NULL, .wifiPassword = NULL };
+    int64_t ret = json_obj_parse(
+        fcontent, strlen(fcontent) + 1, full_json_descr, ARRAY_SIZE(full_json_descr), &parsed_json);
+    if (ret != expected_return_code) {
+        LOG_ERR("JSON Parse Error: %lld", ret);
+        return -1;
+    }
+    if (!parsed_json.deviceID) {
+        LOG_ERR("Parsed JSON is missing the deviceID field.");
+        return -1;
+    }
+    if (!parsed_json.credentialSecret) {
+        LOG_ERR("Parsed JSON is missing the credentialSecret field.");
+        return -1;
+    }
+    if (!parsed_json.wifiSsid) {
```

Comment:
Is this should be parsed only if  is defined over config, like?

```suggestion
    #if defined(CONFIG_WIFI)
       if (!parsed_json.wifiSsid) {
       ....
```